dfa/src/main/scala/dfa/DFA.scala

```diff
@@ -1,3 +1,23 @@
 package dfa // leave this line in the file
 
-// TODO: replace this comment with your implementation
+case class State(label: String)
+
+case class Transition(from: State, to: State, symbol: Char)
+
+class DFA(
+    val states: Set[State],
+    val transitions: Set[Transition],
+    val start: State,
+    val accept: Set[State]):
+
+    def accepts(input: String): Boolean = 
+        // First we set the current state to the start
+        var currentState: State = start
+        for (currentSymbol <- input)
+            // Get transition where from == current state and symbol == current symbol
+            // Then call head to get extract value from set and set the current state to the to State
+            currentState = (transitions.filter(t =>(t.from == currentState && t.symbol == currentSymbol)).head.to)
```

Comment:
I like how concise this is! Our solution was very similar, except we did the .head.to after a check that the filter only found one transition (any more or less and we'd have an invalid DFA). But that probably wasn't necessary to make it all work!